zeppelin-zengine/src/test/java/org/apache/zeppelin/notebook/ParagraphTest.java

```diff
@@ -95,36 +95,53 @@ public void replSingleCharName() {
     Note note = createNote();
     Paragraph paragraph = new Paragraph(note, null, interpreterFactory);
     paragraph.setText("%r a");
-    assertEquals("r", paragraph.getReplText());
+    assertEquals("r", paragraph.getIntpText());
     assertEquals("a", paragraph.getScriptText());
   }
 
   @Test
+  public void replInvalid() {
+    Note note = createNote();
+    Paragraph paragraph = new Paragraph(note, null, interpreterFactory);
+    paragraph.setText("foo %r");
+    assertEquals("", paragraph.getIntpText());
+    assertEquals("foo %r", paragraph.getScriptText());
+
+    paragraph.setText("foo%r");
+    assertEquals("", paragraph.getIntpText());
+    assertEquals("foo%r", paragraph.getScriptText());
+
+    paragraph.setText("% foo");
+    assertEquals("", paragraph.getIntpText());
+    assertEquals("% foo", paragraph.getScriptText());
+  }
+
+  @Test
   public void replNameEndsWithWhitespace() {
     Note note = createNote();
     Paragraph paragraph = new Paragraph(note, null, interpreterFactory);
     paragraph.setText("%test\r\n###Hello");
-    assertEquals("test", paragraph.getReplText());
+    assertEquals("test", paragraph.getIntpText());
```

Comment:
not sure if it matters, how about some tests after a newline, if there isn't any?
like `\r\n%test` (which I think is valid) `%\r\ntest`

Reply:
Good point, more test is added.